### PR TITLE
Add documentation on Lombok usage in Spring Boot projects

### PR DESCRIPTION
**Pull request by Awambeng**

### CODE REVIEW

**Comment by Awambeng** [comment]
@stephane-segning pr for [Research and explain Lombok #84](https://github.com/ADORSYS-GIS/e2e-banking-app/issues/84)

**Comment by Awambeng** [comment]
@stephane-segning work done

File: docs/lombok_guide.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    private int id;
+    private String name;
+    @Setter(AccessLevel.PROTECTED)
+    private String surname;
+}
+```
+
+- With Java Vanilla:
+```java
+public class User {
```
Comment (stephane-segning) [inline docs/lombok_guide.md:96]: Take an example with contextual values (User vs Author)

**Review by stephane-segning — CHANGES_REQUESTED** [review]

**Review by stephane-segning — APPROVED** [review]
Good work

**Comment by Awambeng** [comment]
Thanks prof @stephane-segning

**Pull request merged**